### PR TITLE
[#2314] fix(postgresql): postgresql `public` schema should not be filtered in `listDatabases`

### PR DESCRIPTION
**Pull request by lw-yang**
### What changes were proposed in this pull request?

Remove the `public` schema from the `SYS_PG_DATABASE_NAMES` attribute

### Why are the changes needed?

public schema is the default schema if not specifying schema names. we shouldn't filter it

Fix: #2314

### Does this PR introduce _any_ user-facing change?
N/A.

### How was this patch tested?
Existing tests can cover it.

### CODE REVIEW

**Comment by FANNG1** [comment]
@lw-yang could you fix the failed IT?
```

TestMultipleJdbcLoad > testCreateMultipleJdbc() FAILED
    org.opentest4j.AssertionFailedError: expected: <0> but was: <1>
        at app//org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at app//org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:150)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:145)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:527)
        at app//com.datastrato.gravitino.integration.test.catalog.jdbc.TestMultipleJdbcLoad.testCreateMultipleJdbc(TestMultipleJdbcLoad.java:125)
```

**Comment by jerryshao** [comment]
I think there lacks a UT to cover the changes.

**Comment by lw-yang** [comment]
@FANNG1 I'll fix it soon

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/TestMultipleJdbcLoad.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ public void testCreateMultipleJdbc() throws URISyntaxException {
         postgreSqlCatalog
             .asSchemas()
             .listSchemas(Namespace.of(metalakeName, postgreSqlCatalogName));
-    Assertions.assertEquals(0, nameIdentifiers.length);
+    Assertions.assertNotEquals(0, nameIdentifiers.length);
```
Comment (FANNG1) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/TestMultipleJdbcLoad.java:125]: we should check the schemas only include "public" schema if nameIdentifiers.length > 0

**Comment by FANNG1** [comment]
Could you add a `listSchemaTest` in `CatalogPostgreSqlIT`, add a check whether `public` schema is listed out.

**Comment by lw-yang** [comment]
@FANNG1 i will add a `listSchemaTest` in `CatalogPostgreSqlIT`

**Review by jerryshao — APPROVED** [review]
LGTM. Thanks @lw-yang for your contribution.

**Pull request merged**